### DISCUSSION TITLE
Hiccups when following docker install instructions

### DISCUSSION THREAD

**Discussion by kubycsolutions** [automatic-ripping-machine/automatic-ripping-machine#1136] [Q&A] [answered]
1) Could/should the start script pick up the UID and GID automatically, using 

    -e ARM_UID="$(id -u arm)"

2) Sanity check: On my Intel i3 system, which only has two CPUs, cpuset-cpus should be '2', right?

3) When I launch the docker container using the start script, it seems to be endlessly restarting -- I'll see "Restarting (1) 29 seconds ago" counting up to about a minute, then "Up Less than a second (health: starting)", then it goes back into Restarting. There's nothing in /home/arm/logs. Any tips on how to track down what it's upset about would be welcome.

Test configuration is:

docker run -d \
    -p "8080:8080" \
    -e ARM_UID="1001" \
    -e ARM_GID="1001" \
    -v "/home/arm:/home/arm" \
    -v "/home/arm/cds:/home/arm/music" \
    -v "/home/arm/logs:/home/arm/logs" \
    -v "/home/arm/dvds:/home/arm/media" \
    -v "/home/arm/config:/etc/arm/config" \
    --device="/dev/sr0:/dev/sr0" \
    --device="/dev/sr1:/dev/sr1" \
    --privileged \
    --restart "always" \
    --name "automatic-ripping-machine" \
    --cpuset-cpus='2' \
    automaticrippingmachine/automatic-ripping-machine:latest

4) Just for sanity-check: It should be possible to run this alongside Kodi or similar media player, though not necessarily at the same time due to resource conflict, right? Or would I need to manually switch between loading and player modes?

**Comment by SylvainMT** [comment, marked as answer]
> 1. Could/should the start script pick up the UID and GID automatically, using
> 
> -e ARM_UID="$(id -u arm)"

From what I know of bash scripting, this should work, but it would add a layer of complexity making it more confusing to users who are not familiar with BASH.

> 2. Sanity check: On my Intel i3 system, which only has two CPUs, cpuset-cpus should be '2', right?

"cpuset-cpus", identifies each core that you wish to make available to arm.  Setting it to "2" means you are only making the third core available.  The first core is 0, the second is 1, the third is 2 and so on.  The other caveat here is that if you have a multithreaded system, your 2-core CPU has 4 threads and "cpuset-cpus" treats each thread as a logical core.  For me to give you a better idea which should be your configuration, I would need to know which is Intel i3 you have.

> 3. When I launch the docker container using the start script, it seems to be endlessly restarting -- I'll see "Restarting (1) 29 seconds ago" counting up to about a minute, then "Up Less than a second (health: starting)", then it goes back into Restarting. There's nothing in /home/arm/logs. Any tips on how to track down what it's upset about would be welcome.

Might be fixed by addressing issue number 2.  Checking the logs created by Docker might be a good place to start looking for the error.

> 4. Just for sanity-check: It should be possible to run this alongside Kodi or similar media player, though not necessarily at the same time due to resource conflict, right? Or would I need to manually switch between loading and player modes?

I would be sure to stop the ARM container before starting the Kodi container and vice-versa

**Comment by kubycsolutions** [comment]
Sorry about the delayed response: It's an i3-4130. lscpu reports 

```Architecture:             x86_64
  CPU op-mode(s):         32-bit, 64-bit
  Address sizes:          39 bits physical, 48 bits virtual
  Byte Order:             Little Endian
CPU(s):                   4
  On-line CPU(s) list:    0-3
Vendor ID:                GenuineIntel
  Model name:             Intel(R) Core(TM) i3-4130 CPU @ 3.40GHz
    CPU family:           6
    Model:                60
    Thread(s) per core:   2
    Core(s) per socket:   2
    Socket(s):            1
    Stepping:             3
    CPU(s) scaling MHz:   100%
    CPU max MHz:          3400.0000
    CPU min MHz:          800.0000
    BogoMIPS:             6784.81
```
So per your comments, what I probably want is `--cpuset-cpus='2,3'`, so logical core 0 is left available for hypervisor. Right? Or should I give it 1 as well?

With 1,2,3 it isn't immediately restarting after the first minute; good sign...
